nilearn/plotting/js_plotting_utils.py

```diff
@@ -196,13 +196,11 @@ def colorscale(cmap, values, threshold=None, symmetric_cmap=True, vmax=None):
                       'setting symmetric_cmap to True')
         symmetric_cmap = True
     if vmax is None:
-        if symmetric_cmap:
-            vmax = abs_values.max()
-            vmin = - vmax
-        else:
-            vmin, vmax = values.min(), values.max()
+        vmax = abs_values.max()
+    if symmetric_cmap:
+        vmin = - vmax
     else:
-        vmin = -vmax if symmetric_cmap else 0
+        vmin = values.min()
```

Comment:
In general, I prefer the option of passing an explicit vmin argument, as this is how it's done in matplotlib.

I am not sure what is the state of this in nilearn plotting: I suspect that we have uneven support for vmin, but that might be a usability bug (I think that I was fighting with it a couple days ago, actually). We should fix it and provide a good support for vmin.

I vote for at least testing for the presence of vmin in the kwargs of the calling function, honoring it if symetric_cmap is false, and not honoring it is symetric_cmap is true, but warning about it.

Reply:
https://github.com/nilearn/nilearn/pull/1766#issuecomment-431857095